### PR TITLE
Feature: ESLint integration

### PR DESCRIPTION
**Pull request by Brankub**
ESLint is more flexible and configurable tool for JavaScript linting. I propose to use it in Aldryn Boilerplate. 

*Pros:*
- Flexible: any rule can be toggled, and many rules have extra settings that can be tweaked
- Very extensible and has many plugins available
- Easy to understand output
- Includes many rules not available in other linters, making ESLint more useful for detecting problems
- Best ES6 support, and also the only tool to support JSX
- Supports custom reporters

*Cons:*
- Some configuration required
- Slow, but not a hindrance

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,209 @@
+{
+    "env": {
+        "browser": true,
+        "jquery": true,
+        "es6": true
+    },
+    "globals": {
+        "casper": true
```
Comment (vxsx) [inline .eslintrc.json]: empty

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,209 @@
+{
+    "env": {
+        "browser": true,
+        "jquery": true,
+        "es6": true
```
Comment (vxsx) [inline .eslintrc.json]: not for now :(

Comment (Brankub) [inline .eslintrc.json]: :+1:

File: .eslintrc.json (hunk 1 above)
Comment (Brankub) [inline .eslintrc.json]: :+1:

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        "no-unexpected-multiline": 2,
+        "no-unreachable": 2,
+        "use-isnan": 2,
+        "valid-jsdoc": [2, { "requireReturn": false }],
+        "valid-typeof": 2,
+
+        // Best Practices
+        "accessor-pairs": 2,
+        "block-scoped-var": 2,
+        "complexity": [2, 5],
```
Comment (vxsx) [inline .eslintrc.json]: would probably raise it to 7-10
https://en.wikipedia.org/wiki/Cyclomatic_complexity#Limiting_complexity_during_development

Comment (vxsx) [inline .eslintrc.json]: as it's the "generic" version

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        "browser": true,
+        "node": true,
+        "jquery": true
+    },
+    "globals": {
+      "process.env": true
+    },
+    "root": true,
+    "rules": {
+        // Possible Errors
```
Comment (vxsx) [inline .eslintrc.json]: it's funny that this is invalid json, but the parser is ok with it

File: package.json [hunk 5]
```diff
@@ -1,41 +1,40 @@
...
+    "gulp-sass": "^2.0.4",
+    "gulp-sourcemaps": "^1.5.2",
+    "gulp-util": "^3.0.6",
+    "minimist": "^1.1.3"
+  },
+  "devDependencies": {
+    "browser-sync": "^2.8.2",
+    "browserslist-saucelabs": "^0.1.4",
+    "gulp-cached": "^1.1.0",
+    "gulp-eslint": "^1.1.1",
```
Comment (vxsx) [inline package.json]: duplicated?

Comment (Brankub) [inline package.json]: :-/ strange, will fix it

File: .eslintrc.json (hunk 4 above)
Comment (Brankub) [inline .eslintrc.json]: I think we will stay with it

File: .eslintrc.json (hunk 3 above)
Comment (Brankub) [inline .eslintrc.json]: so, you propose to ues [7, 10]?

**Comment by FinalAngel** [comment]
What does "slow" mean @Brankub compare to the old system? @vxsx can we easily also adapt it then on django-cms?

**Comment by Brankub** [comment]
@FinalAngel "slow" means a little bit slower than JSHint. But it's just a feeling, do not see a huge difference.

File: .eslintrc.json (hunk 3 above)
Comment (vxsx) [inline .eslintrc.json]: eh, [2, 10]

Comment (Brankub) [inline .eslintrc.json]: :-D okay

**Comment by vxsx** [comment]
@FinalAngel i have a wip branch https://github.com/divio/django-cms/compare/develop...FinalAngel:feature/eslint-integration but not all the lint errors are fixed

**Comment by FinalAngel** [comment]
Ideally switching the linter should not throw any linting issues.

**Comment by Brankub** [comment]
@FinalAngel do you mean 2 warnings about console? After discussion with @vxsx was decided to left warnings about such things.

**Comment by vxsx** [comment]
@Brankub he means django-cms, but we'll take it offline :)

**Comment by vxsx** [comment]
LGTM

File: CHANGELOG.rst [hunk 6]
```diff
@@ -6,8 +6,10 @@ Aldryn Boilerplate Bootstrap 3
 =====
 - remove outdated ``keyframes`` mixin (use plain property with autoprefixer)
 - remove ``micro-clearfix`` mixin, use ``clearfix`` class or mixin from bootstrap
+- changed JavaScript linting to ESLint instead of JSHint
 - improved ``docs`` output
 
+>>>>>>> develop
```
Comment (vxsx) [inline CHANGELOG.rst]: oops missed this

Comment (Brankub) [inline CHANGELOG.rst]: Me too, shall I create PR for this?

**Pull request merged**